runtime/src/accounts_db.rs

```diff
@@ -1014,6 +1015,16 @@ struct CleanKeyTimings {
     dirty_ancient_stores: usize,
 }
 
+/// Fundamental details of an account
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct AccountDetails {
```

Comment:
this could be a view over the `LoadedAccount` rather than doing all the copying

```rust
pub struct AccountDetails<'a> {
    pub pubkey: &'a Pubkey,
    ...
}
```

Reply:
Good call out - in light of the request to include data, I'm making that change as the copying got ALOT heavier

Reply:
With current impl., we can't reference the `LoadedAccount`s past those closure that we pass into the below scan function:
https://github.com/solana-labs/solana/blob/2448486106f738c3bd66940babfbf1eb09a56d59/runtime/src/accounts_db.rs#L4890-L4895

So, I had to copy the Pubkey and Hash, but I'm able to avoid doing a copy on rest of account by converting to an `AccountSharedData`

Reply:
As mentioned above, we can't escape the closures without some rework that I don't think this feature merits.

Additionally, it is more desirable to collect all the accounts so we can sort by pubkey to match the ordering used to calculate the accounts delta hash. This rules out the option of streaming as we go.

So, going to mark this as resolved